File: src/main/java/com/yellowsunn/simpleforum/domain/comment/Comment.java
package com.yellowsunn.simpleforum.domain.comment;

import com.yellowsunn.simpleforum.domain.BaseCreatedTimeEntity;
import com.yellowsunn.simpleforum.domain.posts.Posts;
import com.yellowsunn.simpleforum.domain.user.User;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Getter
@NoArgsConstructor
@Entity
public class Comment extends BaseCreatedTimeEntity {

    @Id @GeneratedValue
    @Column(name = "comment_id")
    private Long id;

    @Column(length = 1000, nullable = false)
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Comment parent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private Posts post;

    @Builder
    public Comment(String content, Comment parent, User user, Posts post) {
        this.content = content;
        this.parent = parent;
        this.user = user;
        this.post = post;
    }
}

File: board-service/libs/board-core/src/main/kotlin/com/yellowsunn/boardservice/command/domain/comment/Comment.kt
package com.yellowsunn.boardservice.command.domain.comment

import com.yellowsunn.boardservice.command.domain.BaseTimeEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Comment(
    val articleId: Long,
    val userId: Long,
    content: String?,
    imageUrl: String? = null,
    val parentCommentId: Long? = null,
    var baseCommentId: Long = -1L,
) : BaseTimeEntity() {
    @Id
    @Column(name = "comment_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L

    var content: String? = content
        private set

    var imageUrl: String? = imageUrl
        private set

    var likeCount: Long = 0L
        private set

    fun changeBaseCommentId(baseCommentId: Long) {
        this.baseCommentId = baseCommentId
    }

    fun isReply(): Boolean {
        return parentCommentId != null
    }
}
